catchup/service.go

```diff
@@ -198,6 +204,13 @@ func (s *Service) fetchAndWrite(fetcher rpcs.Fetcher, r basics.Round, prevFetchC
 
 		// Check that the block's contents match the block header (necessary with an untrusted block because b.Hash() only hashes the header)
 		if !block.ContentsMatchHeader() {
+			// Check if this mismatch is due to an unsupported protocol version
+			if _, ok := config.Consensus[block.BlockHeader.CurrentProtocol]; !ok {
+				s.log.Errorf("fetchAndWrite(%v): unsupported protocol version detected", )
```

Comment:
that won't compile; but beside that, we want to also log which protocol is not supported.